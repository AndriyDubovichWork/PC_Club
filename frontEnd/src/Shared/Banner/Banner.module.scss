.Banner {
  background-color: #2d2d2d;
  padding: 5vh 0;

  margin: 5vh 0;

  color: #fff;
  text-align: center;
  .subTitle {
    font-weight: 400;
    .HighLighted {
      color: #ff4d30;
    }
  }
}
