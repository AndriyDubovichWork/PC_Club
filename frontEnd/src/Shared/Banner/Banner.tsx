import React from 'react';
import style from './Banner.module.scss';

const Banner = () => {
  return (
    <div className={style.Banner}>
      <h1 className={style.Title}>Save big with our cheap PC Club</h1>
      <h3 className={style.subTitle}>
        Top PCs. Global Suppliers.
        <span className={style.HighLighted}> 24/7</span> Support.
      </h3>
    </div>
  );
};

export default Banner;
