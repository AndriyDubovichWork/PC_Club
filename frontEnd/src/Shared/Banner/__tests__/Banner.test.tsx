import { render, screen, cleanup } from '@testing-library/react';
import Banner from '../Banner';
import renderer from 'react-test-renderer';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

test('Banner match snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter initialEntries={['/PC_Club/']}>
        <Routes>
          <Route path='/PC_Club/' element={<Banner />} />
        </Routes>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('Banner contains text', () => {
  render(
    <MemoryRouter initialEntries={['/PC_Club/']}>
      <Routes>
        <Route path='/PC_Club/' element={<Banner />} />
      </Routes>
    </MemoryRouter>
  );

  expect(
    screen.getByText('Save big with our cheap PC Club')
  ).toBeInTheDocument();
});
