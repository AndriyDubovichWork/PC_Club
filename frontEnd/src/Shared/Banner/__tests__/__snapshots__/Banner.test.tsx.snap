// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner match snapshot 1`] = `
<div
  className="Banner"
>
  <h1
    className="Title"
  >
    Save big with our cheap PC Club
  </h1>
  <h3
    className="subTitle"
  >
    Top PCs. Global Suppliers.
    <span
      className="HighLighted"
    >
       24/7
    </span>
     Support.
  </h3>
</div>
`;
